crates/miden-objects/src/batch/proven_batch.rs

```diff
@@ -28,8 +29,13 @@ impl ProvenBatch {
     // --------------------------------------------------------------------------------------------
 
     /// Creates a new [`ProvenBatch`] from the provided parts.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the batch expiration block number is not greater than the reference
+    /// block number.
     #[allow(clippy::too_many_arguments)]
-    pub fn new_unchecked(
+    pub fn new(
```

Comment:
As far as I can tell, this constructor does not guarantee that the provided parameters are internally consistent. For example, we don't check that the `BatchId` was computed correctly for this batch. This also makes deserialization "trusted".

This is probably fine for now, but let's create an issue (or discussion) to figure out if we want to change this (at the very least we should try to make the methodology consistent across `ProvenTransaction`, `ProvenBatch`, and `ProvenBlock`).